### PR TITLE
[RFR] Added logIndex property to events

### LINKED ISSUE #653: [SUBGRAPH] Add `logIndex` property to events
**Why?**

Just seems like very basic information that should be available. Currently, the log index is parseable from the Subgraph ID of events... but that's very hacky way of retrieving it.

**How?**

Should be pretty simple change to common mapping logic of events in `/packages/subgraph`.

### PR DESCRIPTION
**Pull request by msoni89**

### CODE REVIEW

File: packages/subgraph/test/helpers/testers.ts [hunk 1]
```diff
@@ -766,6 +770,7 @@ async function fetchIDAEventsAndValidate(
         throw new Error("You have entered the wrong type.");
     }
 
+    console.log(eventDataToValidate)
```
Comment (kasparkallas) [inline packages/subgraph/test/helpers/testers.ts]: console.log

**Review by kasparkallas — COMMENTED** [review]
It makes sense to do the SDK-core changes here as well. Right, @0xdavinchee?

File: packages/subgraph/test/helpers/testers.ts (hunk 1 above)
Comment (msoni89) [reply packages/subgraph/test/helpers/testers.ts]: removed

File: packages/subgraph/test/helpers/helpers.ts [hunk 2]
```diff
@@ -371,15 +371,18 @@ export const modifyFlowAndReturnCreatedFlowData = async (
     const block = await data.provider.getBlock(txnResponse.blockNumber);
     const timestamp = block.timestamp;
     await waitUntilBlockIndexed(txnResponse.blockNumber);
-
-    const { flowRate, deposit } = await data.framework.cfaV1.getFlow({
+    const transactionReciept = await txnResponse.wait();
```
Comment (0xdavinchee) [inline packages/subgraph/test/helpers/helpers.ts]: typo mr

File: packages/subgraph/test/helpers/helpers.ts [hunk 3]
```diff
@@ -426,7 +429,11 @@ export const updateFlowOperatorPermissions = async (
     const block = await data.provider.getBlock(txnResponse.blockNumber);
     const timestamp = block.timestamp;
     await waitUntilBlockIndexed(txnResponse.blockNumber);
-    return { timestamp, txnResponse };
+    const transactionReciept = await txnResponse.wait();
```
Comment (0xdavinchee) [inline packages/subgraph/test/helpers/helpers.ts]: mr mr

File: packages/subgraph/test/helpers/testers.ts [hunk 4]
```diff
@@ -617,26 +617,30 @@ async function executeIDATransactionByTypeAndWaitForIndexer(
...
     }
-
     if (!txnResponse.blockNumber) {
         throw new Error("No block number");
     }
-
     const block = await provider.getBlock(txnResponse.blockNumber);
     await waitUntilBlockIndexed(txnResponse.blockNumber);
-
+    const transactionReciept = await txnResponse.wait();
```
Comment (0xdavinchee) [inline packages/subgraph/test/helpers/testers.ts]: uno mas

File: packages/subgraph/test/validation/eventValidators.ts [hunk 5]
```diff
@@ -34,8 +32,8 @@ export const validateData = <T>(
 ) => {
     const propertiesToValidate = Object.keys(expectedData);
     for (let i = 0; i < propertiesToValidate.length; i++) {
-        expect((queriedData as any)[propertiesToValidate[i]]).to.eql(
-            expectedData[propertiesToValidate[i]],
+        expect(String((queriedData as any)[propertiesToValidate[i]])).to.eql(
```
Comment (0xdavinchee) [inline packages/subgraph/test/validation/eventValidators.ts:35]: same thing w/ .toString() here

**Review by 0xdavinchee — APPROVED** [review]
LGTM

**Pull request merged**